byzcoin/service.go

```diff
@@ -1351,7 +1351,11 @@ func (s *Service) catchupAll() error {
 		// Get the latest block known by the Cothority.
 		reply, err := cl.GetUpdateChain(sb.Roster, sb.Hash)
 		if err != nil {
-			return xerrors.Errorf("getting chain: %v", err)
+			// Might be that the other nodes are not yet up,
+			// so just continue with the other chains.
+			// Call to s.catchup will probably also fail, so skip it.
+			log.Errorf("couldn't get update chain: %+v", err)
+			continue
```

Comment:
If it is unable to retrieve any chain, it will still return `nil`, maybe return a specific error?

Reply:
That will fail in the scenario where all nodes are down, and you want to restart them. You will not get the network up and running, except if you manage to have everybody start their node at exactly the same moment. This is why I changed this to simply continue, which will allow for the other nodes to come up. If there is a disagreement of which is the latest block, the catchup-procedure will be restarted later.

Reply:
More precise to your question: it's not about retrieving a chain, but updating it. This code wants to make sure that the node has the latest available block. But in the above scenario, this is just not possible, and will make it so that the network will fail.

Found it in one of the integration tests ;)